modules/programs/fish.nix

```diff
@@ -274,12 +392,50 @@ in {
       '';
     }
     {
-      xdg.configFile = mapAttrs' (fName: fBody: {
-        name = "fish/functions/${fName}.fish";
+      xdg.configFile = mapAttrs' (name: def: {
```

Comment:
Would not something like
```
xdg.configFile = flip mapAttrs' cfg.functions (name: def: {
  name = "fish/functions/${name}.fish";
  value = {
    text = let
      modifierStr = n: v: optional (v != null) ''--${n}="${toString v}"'';
      modifierBool = n: v: optional (v != null && v) "--${n}";

      mods = with def;
        modifierStr "description" description ++ modifierStr "wraps" wraps
        ++ modifierStr "on-event" onEvent
        ++ modifierStr "on-variable" onVariable
        ++ modifierStr "on-job-exit" onJobExit
        ++ modifierStr "on-process-exit" onProcessExit
        ++ modifierStr "on-signal" onSignal
        ++ modifierBool "no-scope-shadowing" noScopeShadowing
        ++ modifierStr "inherit-variable" inheritVariable
        ++ modifierStr "argument-names" argumentNames;

      modifiers = if isAttrs def then " ${toString mods}" else "";
      body = if isAttrs def then def.body else def;
    in ''
      function ${name}${modifiers}
        ${body}
      end
    '';
  };
});
```
work?

Reply:
Indeed it does! That looks beautiful.

The only problem is: `argumentNames` should not be quoted (otherwise it becomes `--argument-names="asdf fdsa"` which is a variable named `asdf fdsa`, which probably isn't expected).

I just added a `modifierStrs` function and removed the quotes from it.